cc3d/player5/Plugins/ViewManagerPlugins/SimpleTabView.py

```diff
@@ -2346,17 +2350,51 @@ def restore_default_settings(self):
         if not self.simulationIsRunning:
             return
 
-        Configuration.replace_custom_settings_with_defaults()
+        ret = QMessageBox.question(self, "Restoring Default Simulation Setting",
+                                    "Need to stop existing simulation to restore default simulation settings. "
+                                    "Do you want to proceed?",
+                                   QMessageBox.Yes | QMessageBox.No
+                                   )
+        if ret == QMessageBox.No:
+            return
+
+        self.restore_default_settings_local_flag = True
+        self.__stopSim()
+        # Configuration.replace_custom_settings_with_defaults()
 
-    @staticmethod
-    def restore_default_global_settings():
+
+    def restore_default_global_settings(self):
         """
         Removes global settings
 
         :return: None
         """
 
+        if self.simulationIsRunning:
+            ret = QMessageBox.question(self, "Restoring Global Simulation Setting",
+                                       "Need to <b>stop</b> existing simulation and <b>close the Player</b>"
+                                       " to restore default global settings. <br/>"
+                                       "Do you want to proceed?",
+                                       QMessageBox.Yes | QMessageBox.No
+                                       )
+            if ret == QMessageBox.No:
```

Comment:
I think this handles cancel (escape key) as well, so it's fine. Correct me if I'm wrong.